alibi_detect/cd/metrics.py

```diff
@@ -0,0 +1,333 @@
+        a longer runtime.
+    n_samples
+        The number of samples to randomly sample from the data reservoir to create reference and test data for each run.
+        The two data subsampled data sets will each contain `n_samples` instances, such that `2*n_samples` instances
+        are sampled in total.
+    qq_plot
+        Whether to visualise the p-values on a Quantile-Quantile plot. This can also be done later with the
+        :func:`~plot_qq` function.
+    hist_plot
+        Whether to visualise the p-values on a histogram. This can also be done later with the :func:`~plot_hist`
+        function.
+    random_seed
+        Random seed (or :class:`~np.random.Generator`) to use for random sampling. If `None`, then fresh,
+        unpredictable entropy will be pulled from the OS.
+
+    Returns
+    -------
+    A numpy array containing the detector's p-value prediction for each of the `n_runs` number of runs.
+    """
+    # Check data sizes
+    n_data = len(X)
+    _check_sufficient_data_size(X, n_samples, n_runs)
+
+    # Setup detector_kwargs
+    if detector_kwargs is None:
+        detector_kwargs = {}
+
+    # Perform preprocessing (if preprocess_fn exists) to save repeated compute in loop
+    preprocess_fn = detector_kwargs.pop('preprocess_fn', None)
+    if preprocess_fn is not None:
+        X = preprocess_fn(X)
+
+    # NumPy RNG
+    rng = np.random.default_rng(random_seed)
+
+    # Main experiment loop
+    p_vals_list = []
+    for _ in tqdm(range(n_runs)):
+        # Subsample data
+        idx = rng.choice(n_data, size=n_data, replace=False)
+        idx_ref, idx_nodrift = idx[:n_samples], idx[n_samples:2*n_samples]
+        x_ref, x_nodrift = X[idx_ref], X[idx_nodrift]
+        detector_args, predict_args = [x_ref], [x_nodrift]
+
+        if C is not None:
+            c_ref, c_nodrift = C[idx_ref], C[idx_nodrift]
+            detector_args.append(c_ref)
+            predict_args.append(c_nodrift)
+
+        if model is not None:
+            detector_args.append(model)
+
+        # Init detector and predict
+        dd = detector(*detector_args, **detector_kwargs)
+        preds = dd.predict(*predict_args)
+        p_vals_list.append(preds['data']['p_val'])
+    p_vals = np.array(p_vals_list).flatten()
+    # TODO - is above OK? (to flatten over features for univariate detectors)
+
+    # QQ-plot
+    if qq_plot:
+        plot_qq(p_vals)
+    # Histogram plot
+    if hist_plot:
+        plot_hist(p_vals)
+    # Show plots
+    if qq_plot or hist_plot:
+        plt.show()
+
+    return p_vals
+
+
+def eval_test_power(
+    detector: type,  # TODO: type w/ BaseDriftDetector if introduced.
+    X_ref: Union[np.ndarray, list],
+    X_drift: Union[np.ndarray, list],
+    C_ref: Optional[np.ndarray] = None,
+    C_drift: Optional[np.ndarray] = None,
+    model: Optional[Callable] = None,
+    detector_kwargs: Optional[dict] = None,
+    sig_levels: np.ndarray = np.linspace(0.05, 0.5, 10),
```

Comment:
Not sure about this as a default. Maybe something more similar to np.linspace(0.01, 0.1, 10)?